File: programmers/Summer-Winter Coding/2018_skillTree.js
// 2021.04.14. Summer-Winter coding(2018)
// Skill tree

function solution(skill, skill_trees) {
    let answer = 0;
    
    for(let i=0;i<skill_trees.length;i++){
        const skillArray = skill.split('');
        
        for(let j=0 ; j<skill_trees[i].length ; j++){
            if (skillArray.includes(skill_trees[i][j])){
                if (skill_trees[i][j] !== skillArray.shift()){
                    break;
                }
            }
            if (j===skill_trees[i].length-1){
                answer += 1;
            }
        }
    }
    return answer;
}

console.log(solution('CBD',["BACDE", "CBADF", "AECB", "BDA"]))

File: programmers/Summer-Winter Coding/2018_skill-tree.py
# 2021.04.14. Summer-Winter coding(2018)
# Skill tree

def solution(skill, skill_trees):
    answer = 0
    
    for skills in skill_trees:
        skill_list = list(skill)
        # using python's for-else statement
        for s in skills:
            if s in skill:
                if s != skill_list.pop(0):
                    break
        else:answer += 1

    return answer


print(solution('CBD', ["BACDE", "CBADF", "AECB", "BDA"]	))
